okapi-core/src/main/java/org/folio/okapi/managers/TimerManager.java

```diff
@@ -178,23 +209,33 @@ private void fireTimer(String tenantId, ModuleDescriptor md, TimerDescriptor tim
   }
 
   Future<ModuleDescriptor> getModuleForTimer(String tenantId, String timerId) {
-    return tenantManager.getEnabledModules(tenantId).compose(list -> {
-      String product = timerId.substring(0, timerId.indexOf(TIMER_ENTRY_SEP));
-      for (ModuleDescriptor md : list) {
-        if (product.equals(md.getProduct())) {
-          return Future.succeededFuture(md);
+    return tenantManager.getEnabledModules(tenantId)
+        .compose(list -> getModuleForTimer(list, timerId)
+        ).recover(cause -> Future.succeededFuture(null));
+  }
+
+  Future<ModuleDescriptor> getModuleForTimer(List<ModuleDescriptor> list, String timerId) {
```

Comment:
This should return a `ModuleDescriptor`, not a `Future<ModuleDescriptor>`, because there isn't any async.

Then
`.compose(list -> getModuleForTimer(list, timerId))` can be replaced by
`.map(list -> getModuleForTimer(list, timerId))`